force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -228,18 +287,24 @@ public with sharing class ClassesObjectsMethods {
     public class Book {
         // Declare three public instance variables - title, author, pages
         // title
+        Public String title;
         // author
+        Public String author;
         // pages
+        Public Integer pages;
     
         // Book constructor to initialize the title and author
-        public Book() {
+        public Book(String title, String author, Integer pages) {
             // Initialize the title and author
+            this.title = title;
+            this.author = author;
+            this.pages = pages;
         }
     
         // Method that returns the details of the book
         public String getBookDetails() {
             // return A string containing the details of the book in the format "Title: <title>, Author: <author>"
-            return null; // Replace null with the variable you used to store the result
+            return 'Title: ' + title + ', Author: ' + author + ', Pages: ' + pages; // Replace null with the variable you used to store the result
```

Comment:
When referencing instance variables, it's best to use the `this` keyword (although it's not enforced in Apex).
```suggestion
            return 'Title: ' + this.title + ', Author: ' + this.author + ', Pages: ' + pages; // Replace null with the variable you used to store the result
```